pkg/services/hivelocity/device/device_test.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package device implements functions to manage the lifecycle of Hivelocity devices.
+package device
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hivelocity/cluster-api-provider-hivelocity/api/v1alpha1"
+	"github.com/hivelocity/cluster-api-provider-hivelocity/pkg/scope"
+	mockclient "github.com/hivelocity/cluster-api-provider-hivelocity/pkg/services/hivelocity/client/mock"
+	hv "github.com/hivelocity/hivelocity-client-go/client"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_associateDevice(t *testing.T) {
+	client := mockclient.NewHVClientFactory().NewClient("my-api-key")
+	ctx := context.Background()
+	device, err := client.GetDevice(ctx, mockclient.FreeDeviceID)
+	require.NoError(t, err)
+	s := Service{}
+	s.scope = &scope.MachineScope{}
```

Comment:
Isn't there a NewScope function that is used in controllers folder as well?

Reply:
No, up to now there is no NewScope() func. Should I add it, or leave it like it is?

Reply:
Sorry, I was blind. You are right, there is a NewMachineScope() func. 

You said:

> Isn't there a NewScope function that is used in controllers folder as well?

What is the next step now? Mark this thread as resolved, because the your question was answered?

Or do you think NewMachineScope should get used?

Reply:
I think we should use it!